core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/PropertiesUtils.java

```diff
@@ -624,4 +629,39 @@ public static boolean isDeprecated(Parameter parameter) {
 	private static boolean isParameterValueTrue(String parameterValue) {
 		return parameterValue != null && parameterValue.equalsIgnoreCase("true");
 	}
+	
+	/**
+	 * converts a given duration to milliseconds
+	 * 
+	 * @param value
+	 * @return
+	 */
+	public static boolean validateDuration(String value) {
+		try {
+			ICamelManagerService svc = CamelServiceManagerUtil.getManagerService(getCurrentProjectCamelVersion());
+	    	if (svc == null) return false;
+	    	// try to convert to millis which will throw exception on error
+	    	svc.durationToMillis(value);
+		} catch (IllegalArgumentException ex) {
+			// invalid string
+			return false;
```

Comment:
They are providing more detailed errors in the message of the thrown illegal argument exception, it will improve user feedback if we can provide them to him